### PR TITLE
Windows: Factoring out unused fields

### PR DESCRIPTION
**Pull request by lowenna**
- relates to https://github.com/moby/moby/issues/10662

@swernli

This PR is part of the proposal described in docker/docker issue [10662](https://github.com/moby/moby/issues/10662) to port the docker daemon to Windows. This PR addresses a lot of "TODO Windows" by refactoring many platform specific fields which were still common across all platforms into the right place. This includes 

-- daemon\config: CorsHeaders; EnableCors
-- daemon\container: ResolvConfPath; HostnamePath; HostsPath; UpdateDns; MountPoints; Volumes; VolumesRW
-- daemon\inspect ContainerJSONBase
-- reworking of daemon\create due to above
-- reworking of daemon\volume due to above (there is more to do here, but I'm trying to limit the size of the PR)

### CODE REVIEW

**Comment by calavera** [comment]
It looks good at first glance. I don't know if `createPlatformSpecific` is too abstract.

File: daemon/create_unix.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	"strings"
+
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/docker/docker/runconfig"
+	"github.com/docker/libcontainer/label"
+)
+
+// createPlatformSpecific performs platform specific container create functionality
+func createPlatformSpecific(container *Container, config *runconfig.Config) error {
+
```
Comment (calavera) [inline daemon/create_unix.go]: please, don't leave an empty line here.

File: daemon/inspect_unix.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// +build !windows
+
+package daemon
+
+import "github.com/docker/docker/api/types"
+
+// This sets platform-specific fields
+func setPlatformSpecificContainerFields(container *Container, contJSONBase *types.ContainerJSONBase) *types.ContainerJSONBase {
+
```
Comment (calavera) [inline daemon/inspect_unix.go]: please, remove empty line

**Comment by jhowardmsft** [comment]
I was trying to be concise :) But next push will update it to createContainerPlatformSpecificSettings

File: daemon/create_unix.go (hunk 1 above)
Comment (jhowardmsft) [inline daemon/create_unix.go]: Fixed

File: daemon/inspect_unix.go (hunk 2 above)
Comment (jhowardmsft) [inline daemon/inspect_unix.go]: Fixed

**Comment by jhowardmsft** [comment]
Rebase from hell completed :)

**Comment by calavera** [comment]
oh, gosh. I'm very sorry. Changes LGTM.

**Comment by jfrazelle** [comment]
:( needs rebasse

**Comment by jfrazelle** [comment]
ping me after :)

**Comment by jhowardmsft** [comment]
Rebased. Waiting for CI to complete.

**Comment by jfrazelle** [comment]
ok only seems to be failing on stupid urandom, LGTM, merging

**Pull request merged**